org.adempiere.base/src/org/compiere/model/MLookup.java

```diff
@@ -995,6 +996,148 @@ private synchronized static CCache<String, ValueNamePair> getDirectValueNamePair
 		return vnpCache;
 	}
 	
+
+
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param windowNo
+	 * @param tabNo
+	 * @return null if tableID &lt;= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, int windowNo, int tabNo) {
+		return getRecordsLookup(tableID, false, windowNo, tabNo);
+	}
+	
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param windowNo
+	 * @param tabNo
+	 * @param keyColumn
+	 * @return null if tableID &lt;= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, int windowNo, int tabNo, String keyColumn) {
+		return getRecordsLookup(tableID, false, windowNo, tabNo, keyColumn);
+	}
+
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param tableBased - if the lookup checks for the UUID Key Column based on the table (true), or on the columnName (false)
+	 * @param windowNo
+	 * @param tabNo
+	 * @return null if tableID <= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, boolean tableBased, int windowNo, int tabNo) {
+		return getRecordsLookup(tableID, tableBased, windowNo, tabNo, null);
+	}
+	
+	/**
+	 * Get Lookup
+	 * @param tableID
+	 * @param tableBased - if the lookup checks for the UUID Key Column based on the table (true), or on the columnName (false)
+	 * @param windowNo
+	 * @param tabNo
+	 * @param keyColumn
+	 * @return null if tableID <= 0 or the table doesn't have any key column, else {@link MLookup}
+	 */
+	public static MLookup getRecordsLookup(int tableID, boolean tableBased, int windowNo, int tabNo, String keyColumn) {
+		if(tableID <= 0)	
+			return null;
+		MTable mTable = MTable.get(Env.getCtx(), tableID, null);
+		// load key column
+		if(Util.isEmpty(keyColumn)) {
+			if (tableBased) {
+				String[] keyColumns = mTable.getKeyColumns();
+				keyColumn = keyColumns != null && keyColumns.length > 0 ? keyColumns[0] : null;
+			} else {				
+				keyColumn = PO.getUUIDColumnName(mTable.getTableName());
+			}

```

Comment:
I don't quite get it why do you need an argument (tableBased) for this. The method should be using integer id key column if the table has a single integer id key column and if that's not the case, fallback to uuid key column.

Reply:
I've removed the tableBased argument completely, but I think the keyColumn is needed, since in the RecordEditor you want to tell if you want to use the _UU column as a key - in case of RecordUUEditor you want to set the UU column always. Also this reuses the abstract getKeyColumn method of RecordEditor